include/deal.II/numerics/matrix_tools.h

```diff
@@ -972,7 +980,11 @@ namespace MatrixTools
   /**
    * Exception
    */
-  DeclException0 (ExcBlocksDontMatch);
+  DeclExceptionMsg (ExcBlocksDontMatch,
```

Comment:
This exception could be used (I don't know if it already is) in the more generic context where the block structure of any two objects (matrices, vectors, ...) does not match up. Maybe the message should reflect that instead of being specific to a matrix and a vector?